src/core/socket.c

```diff
@@ -1852,7 +1852,7 @@ static int socket_coldplug(Unit *u) {
                    SOCKET_FINAL_SIGTERM,
                    SOCKET_FINAL_SIGKILL)) {
 
-                r = unit_watch_pid(UNIT(s), s->control_pid);
+                r = unit_watch_pid(UNIT(s), s->control_pid, true);
```

Comment:
should be false (see above)

Reply:
Hmm are you sure about this one ?

s->control_pid was supposed to be owned exclusively by the socket unit when serializing. So why should this have changed when deserializing ?

Reply:
same reasons as above: pids can migrate and become watched by many units that way. only when we just forked something off we can know for sure nobody else considers it relevant right now.

what is key to know is that each unit in systemd watches: 

1. 0 or 1 main pids
2. 0 or 1 contol pids
3. 0, 1 or more pids inside its cgroup

Of these 1 + 2 might also be part of 3, and usually are, but don't have to be. And if they aren't then they are member of another unit's cgroup and thus the same PID might be associated with multiple units.